lib/auth/auth_with_roles.go

```diff
@@ -5416,9 +5416,19 @@ func (a *ServerWithRoles) DeleteMFADevice(ctx context.Context) (proto.AuthServic
 
 // AddMFADeviceSync is implemented by AuthService.AddMFADeviceSync.
 func (a *ServerWithRoles) AddMFADeviceSync(ctx context.Context, req *proto.AddMFADeviceSyncRequest) (*proto.AddMFADeviceSyncResponse, error) {
-	// The token provides its own authorization and authentication.
-	res, err := a.authServer.AddMFADeviceSync(ctx, req)
-	return res, trace.Wrap(err)
+	switch {
+	case req.TokenID != "":
+	default: // ContextUser
+		if a.hasBuiltinRole(types.RoleProxy) {
```

Comment:
Why do we single out Proxy in particular? Would it make sense to disallow other system roles, too?

Reply:
The Proxy identity is often used when the user doesn't have one. An example is the initial login. The Web UI also uses Proxy REST APIs extensively, so this flags silly errors explicitly. Other system roles don't usually call this "family" of RPCs.

Reply:
Sure, but is there any actual reason to allow other system roles to call these RPCs?

Reply:
Sure, let me give it a shot.

Reply:
Done, PTAL.